esp_flasher/serial_console.py

```
@@ -0,0 +1,202 @@
+            self.serial_reader.start()
+            
+            # Enable input
+            self.input_field.setEnabled(True)
+            self.send_button.setEnabled(True)
+            self.input_field.setFocus()
+            
+            self.colored_console.write(f"\033[32mConnected to {port_name} at {baudrate} baud\033[0m\n")
+            
+        except serial.SerialException as e:
+            self.show_error(f"Failed to open serial port: {e}")
+        except Exception as e:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)